llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -891,5 +930,42 @@ getDeviceCodeSplitter(ModuleDesc &&MD, IRSplitMode Mode, bool IROutputOnly,
   return std::make_unique<ModuleCopier>(std::move(MD), std::move(Groups));
 }
 
+SmallVector<ModuleDesc, 2> splitByESIMD(ModuleDesc &&MD,
```

Comment:
It will be helpful to add a comment about the functionality here. You can borrow from the commit message for this.

Thanks